frame/staking/src/migrations.rs

```diff
@@ -20,6 +20,36 @@ use super::*;
 use frame_election_provider_support::SortedListProvider;
 use frame_support::traits::OnRuntimeUpgrade;
 
+pub mod v11 {
+	use super::*;
+	use frame_support::{pallet_prelude::ValueQuery, storage_alias};
+
+	#[storage_alias]
+	type HistoryDepth<T: Config> = StorageValue<Pallet<T>, u32, ValueQuery>;
+
+	/// Clean up `HistoryDepth` from storage
+	///
+	/// We will be depending on the configurable value of `HistoryDepth` post
+	/// this release.
+	pub struct MigrateToV11<T>(sp_std::marker::PhantomData<T>);
+	impl<T: Config> OnRuntimeUpgrade for MigrateToV11<T> {
+		fn on_runtime_upgrade() -> frame_support::weights::Weight {
+			if StorageVersion::<T>::get() == Releases::V10_0_0 {
+				// may be just check its greater?
+				debug_assert_eq!(T::HistoryDepth::get(), HistoryDepth::<T>::get());
```

Comment:
should go in `pre_migrate`

Reply:
@kianenigma Do you mean we should fail migration if this condition does not match? Or move as it is (debug assertion) in the pre hook?